### PR TITLE
Fix undefined nft ids

### PR DESCRIPTION
**Pull request by micaelffrancoAV**
What: Fix validateNftId function to throw on undefined nft ids.

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ function validateNftId(nftId) {
     if (isHex(nftId)) return new BN(nftId.substring(2), 16).toString(10);
     else {
       const hexNftId = new BN(nftId).toString(16);
-      if (hexNftId.length <= 64) return nftId;
+      if (nftId && hexNftId.length <= 64) return nftId;
       else throw nftId;
```
Comment (alexpArtos) [inline lib/common.js:166]: So, if `nftId` is undefined we throw an undefined error? Shouldn't we have something else in here to explain that error?

**Review by ivan-cholakov — APPROVED** [review]

File: lib/common.js (hunk 1 above)
Comment (micaelffrancoAV) [reply lib/common.js:166]: We have a catch clause that returns this message:
`catch (nftId) {
    throw new Error(`Invalid nftId type: ${nftId}`);
}`

File: lib/common.js [hunk 2]
```diff
@@ -160,11 +160,10 @@ function validateIsArray(array) {
 function validateNftId(nftId) {
   try {
     if (isHex(nftId)) return new BN(nftId.substring(2), 16).toString(10);
-    else {
-      const hexNftId = new BN(nftId).toString(16);
-      if (hexNftId.length <= 64) return nftId;
-      else throw nftId;
-    }
+    if (!nftId) throw nftId;
+    const hexNftId = new BN(nftId).toString(16);
```
Comment (nahuseyoum) [inline lib/common.js:164]: I would rewrite this as something like:
```
if (!nftId) throw nftId;

const hexNftId = isHex(nftId) ? nftId : new BN(nftId).toString(16);
if (hexNftId.length > 64) throw nftId;

return isHex(nftId) ? new BN(nftId.substring(2), 16).toString(10) : nftId

```

**Review by nahuseyoum — APPROVED** [review]

**Review by alexpArtos — APPROVED** [review]

File: lib/common.js (hunk 2 above)
Comment (fluorostani) [reply lib/common.js:164]: Yep, we agreed the NFT ID would always be decimal (much to my chagrin)

Comment (alexpArtos) [reply lib/common.js]: I'm with you on this, @fluorostani :grin:

**Pull request merged**